<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>731</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="1" column="2">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox
{
background-color: rgb(248, 219, 255);
border:2px solid blue;
}</string>
      </property>
      <property name="title">
       <string>位置控制</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_23">
           <property name="text">
            <string>当前位置</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_pulse_Now"/>
         </item>
         <item>
          <widget class="QLabel" name="label_25">
           <property name="text">
            <string>圈</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_PositionAsk">
           <property name="text">
            <string>当前位置圈数查询</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_24">
           <property name="text">
            <string>编码器反馈值</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_pulse_Now_M"/>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_PositionAskM">
           <property name="text">
            <string>编码器反馈值查询</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_22">
           <property name="text">
            <string>发送定位脉冲数</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_pulseTarget"/>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton_pulseStart">
           <property name="text">
            <string>位置模式启动</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_SetZero">
         <property name="text">
          <string>直接将当前位置设置为零点</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_gotoLimitSwitch">
         <property name="text">
          <string>限位归零（运动到限位开关位置停下来）</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox
{
background-color: rgb(248, 219, 255);
border:2px solid blue;
}</string>
      </property>
      <property name="title">
       <string>速度控制</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>电机ID号</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_canID"/>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>方向(0顺时针)</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_DIR"/>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>转速</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_speedRPM"/>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>加速度</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_AccSpeed"/>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="pushButton_speedStop">
         <property name="text">
          <string>速度立即停止(低速模式使用)</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="pushButton_speeddown">
           <property name="text">
            <string>减速停止</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>减速度</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEdit_DecSpeed"/>
         </item>
        </layout>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QPushButton" name="pushButton_speedUp">
         <property name="text">
          <string>速度模式启动</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox
{
background-color: rgb(248, 219, 255);
border:2px solid blue;
}</string>
      </property>
      <property name="title">
       <string>接口板信息</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>CAN通道数：</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>硬件版本号:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>接口库版本号：</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>中断号:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QLabel" name="label_canChannel">
           <property name="text">
            <string>N/A</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_canHardVersion">
           <property name="text">
            <string>N/A</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_canSoftVersion">
           <property name="text">
            <string>N/A</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_InterruptNum">
           <property name="text">
            <string>N/A</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QGroupBox" name="groupBox_5">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox
{
background-color: rgb(248, 219, 255);
border:2px solid blue;
}</string>
      </property>
      <property name="title">
       <string>发送数据格式设置</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QComboBox" name="comboBox_6">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <item>
          <property name="text">
           <string>标准帧</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>扩展帧</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_5">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <item>
          <property name="text">
           <string>数据帧</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>远程帧</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_4">
         <item>
          <property name="text">
           <string>USB-CAN模块通道0</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>USB-CAN模块通道1</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="font">
       <font>
        <pointsize>12</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox
{
background-color: rgb(248, 219, 255);
border:2px solid blue;
}</string>
      </property>
      <property name="title">
       <string>设置</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>设备类型：</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comboBox">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>USBCAN-2A</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>USBCAN-2C</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>CANalyst-II</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>设备索引：</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="comboBox_2">
           <item>
            <property name="text">
             <string>0</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>3</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>6</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>7</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>9</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>10</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>11</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>12</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>13</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>波特率：</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="comboBox_3">
           <item>
            <property name="text">
             <string>500Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>20Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>40Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>50Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>80Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>100Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>125Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>200Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>250Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>400Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>10Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>666Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>800Kbps</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>1000Kbps</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QPushButton" name="pushButtonOpenCan">
           <property name="text">
            <string>打开设备</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QPushButton" name="pushButton_resetCan">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>复位</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QPushButton" name="pushButton_InitCan">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>初始化CAN</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QPushButton" name="pushButton_startCan">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>启动CAN</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>731</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
